app/src/androidTest/java/org/mozilla/fenix/ui/robots/AddToHomeScreenRobot.kt

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui.robots
+
+import android.net.Uri
+import android.os.Build
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.clearText
+import androidx.test.espresso.action.ViewActions.typeText
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiScrollable
+import androidx.test.uiautomator.UiSelector
+import androidx.test.uiautomator.Until
+import org.hamcrest.CoreMatchers.anyOf
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
+import org.mozilla.fenix.helpers.click
+import org.mozilla.fenix.helpers.ext.waitNotNull
+
+/**
+ * Implementation of Robot Pattern for the Add to homescreen feature.
+ */
+class AddToHomeScreenRobot {
+
+    fun addShortcutName(title: String) =
+        shortcutNameField()
+            .perform(clearText())
+            .perform(typeText(title))
+
+    fun addHomeScreenShortcut(url: Uri, title: String) {
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(url) {
+        }.openThreeDotMenu {
+        }.openAddToHomeScreen {
+            mDevice.waitNotNull(Until.findObject(By.text("Add to Home screen")), waitingTime)
+            addShortcutName(title)
+            addButton().click()
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                addAutomaticallyButton().click()
+            }
+        }
+    }
+
+    class Transition {
+        fun openHomeScreenShortcut(title: String, interact: BrowserRobot.() -> Unit): BrowserRobot.Transition {
+            mDevice.wait(
+                Until.findObject(By.text(title)),
+                waitingTime
+            )
+            mDevice.findObject((UiSelector().text(title))).clickAndWaitForNewWindow()
```

Comment:
Does `clickAndWaitForNewWindow` need a provided timeout value?

Reply:
"Waits for window transitions that would typically take longer than the usual default timeouts" - I don't see anywhere what is the amount of time it waits...
I haven't seen any issues yet without a specified time either, but I could add the TestAssetHelper.waitingTime - would that be better?

Reply:
Yeah I would use that